docs/source/models/drinking_water_model_research.md

```diff
@@ -0,0 +1,977 @@
+
+### Calculating Demand Deficit
+Demand deficit must be calculated manually for Pressure-Dependent Demand (PDD) simulations:
+
+```python
+# Get expected demand
+expected_demand = wntr.metrics.expected_demand(wn)
+
+# Get actual delivered demand from results
+actual_demand = results.node['demand'].loc[:, wn.junction_name_list]
+
+# Calculate deficit
+demand_deficit = expected_demand - actual_demand
+```
+
+### Context
+In PDD simulations, when pressure is insufficient, less water is delivered than requested. The deficit represents this shortfall.
+
+### Reference
+- https://github.com/usepa/WNTR/blob/main/documentation/resultsobject.rst
+
+---
+
+## Question 3: Tank Full but Not Overflowable (Lines 123-127)
+
+**Q:** What happens to a tank when it's full but not overflowable?
+
+**A: EPANET/WNTR stops any inflow when tank reaches max_level.**
+
+### Behavior
+- **overflow=False (default):** Tank acts as a closed boundary at max_level - no more water can enter
+- **overflow=True:** Any input beyond max_level becomes spillage and is removed from the system
+
+### Technical Details
+- Tank is effectively "closed" for inflow when at max_level
+- Outflow is still permitted until tank drops below max_level
+- Similarly, outflow stops at min_level
+
+### Reference
+- http://wateranalytics.org/EPANET/_tanks_page.html
+
+---
+
+## Question 4: min_level, min_volume, max_level Interaction (Lines 148-155)
+
+**Q:** How do min_level, min_volume, max_level work? Can they be combined?
+
+**A: They are mutually exclusive based on tank type.**
+
+### Cylindrical Tanks (constant diameter)
+Use these attributes:
+- `diameter` - tank diameter
+- `min_level` - minimum water level for drainage
+- `max_level` - maximum water level (overflow threshold if enabled)
+
+### Volume-Curve Tanks
+Use these attributes:
+- `vol_curve_name` - references a volume curve defining volume vs. depth
+- `min_vol` - minimum volume for drainage
+
+### Important Notes
+- When `vol_curve_name` is defined, `diameter` becomes irrelevant
+- The `get_volume(level)` method handles both cases automatically
+- **Documentation assumption is correct:** Use one group of attributes OR the other, not both
+
+### Reference
+- https://usepa.github.io/WNTR/apidoc/wntr.network.elements.Tank.html
+
+---
+
+## Question 5: Reservoirs as Drains (Lines 182-186)
+
+**Q:** Can reservoirs be used as water drains? Can they connect directly to pipes?
+
+**A: Yes to both, with caveats.**
+
+### Reservoir as Drain
+- Reservoirs are designed as **fixed-head boundary conditions**
+- If reservoir head is lower than connected nodes, water flows INTO the reservoir (acts as drain)
+- Pressure property is always 0 for reservoirs
+
+### Direct Pipe Connection
+- Reservoirs **CAN connect directly to pipes** - no pump required
+- However: **PRV, PSV, FCV cannot connect directly to reservoir/tank** (use an intermediate pipe)
+
+### Reference
+- https://usepa.github.io/WNTR/apidoc/wntr.network.elements.Reservoir.html
```

Comment:
Did you check this reference? It doesn't mention anything about the answer to the question. Did you verify the answer?